### PR TITLE
Enable Microsoft.CodeAnalysis.FxCopAnalyzers for Jellyfin.Common

### PR DESCRIPTION
**Pull request by Bond-009**
Depends on #2731

### CODE REVIEW

File: MediaBrowser.Common/Net/HttpRequestOptions.cs [hunk 1]
```diff
@@ -112,7 +110,7 @@ public enum CacheMode
     }
 
     [Flags]
-    public enum CompressionMethod
+    public enum CompressionMethods
```
Comment (mark-monteiro) [inline MediaBrowser.Common/Net/HttpRequestOptions.cs:113]: Why rename this? I think enum names are generally supposed to be singular not plural

File: MediaBrowser.Common/Net/HttpResponseInfo.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ namespace MediaBrowser.Common.Net
     /// <summary>
     /// Class HttpResponseInfo.
     /// </summary>
-    public class HttpResponseInfo : IDisposable
+    public sealed class HttpResponseInfo : IDisposable
```
Comment (mark-monteiro) [inline MediaBrowser.Common/Net/HttpResponseInfo.cs:11]: I think just get rid of the `IDisposable` interface instead, the `Dispose()` method here is empty

File: MediaBrowser.Common/Net/HttpRequestOptions.cs (hunk 1 above)
Comment (Bond-009) [reply MediaBrowser.Common/Net/HttpRequestOptions.cs:113]: Not if they are flag enums

File: MediaBrowser.Common/Net/HttpResponseInfo.cs (hunk 2 above)
Comment (Bond-009) [reply MediaBrowser.Common/Net/HttpResponseInfo.cs:11]: That would cause a lot of breakage

Comment (mark-monteiro) [reply MediaBrowser.Common/Net/HttpResponseInfo.cs:11]: Sorry, I was looking at the error pane in VS wrong when I checked this locally. Yeah that would be a much bigger change.

**Pull request merged**